pkg/controllers/member/serviceexport/controller.go

```diff
@@ -256,6 +268,70 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	return ctrl.Result{}, nil
 }
 
+func (r *Reconciler) setAzureRelatedInformation(ctx context.Context, service *corev1.Service, export *fleetnetv1alpha1.InternalServiceExport) error {
+	export.Spec.Type = service.Spec.Type
+	if service.Spec.Type != corev1.ServiceTypeLoadBalancer {
+		return nil
+	}
+	export.Spec.IsInternalLoadBalancer = service.Annotations[objectmeta.AzureLoadBalancerInternalAnnotation] == "true"
+	if export.Spec.IsInternalLoadBalancer {
+		// no need to populate the PublicIPResourceID and IsDNSLabelConfigured which are only applicable for external load balancer
+		return nil
+	}
+	id, err := r.fetchPublicIPResourceID(ctx, service)
+	if err != nil {
+		return err
+	}
+	if id == nil || len(*id) == 0 {
+		return fmt.Errorf("public ip resource id %v has not been configured in the load balancer yet", id)
+	}
+	export.Spec.PublicIPResourceID = id
+	// Note the user can set the dns label via the Azure portal or Azure CLI without updating service.
+	// This information may be stale as we don't monitor the public IP address resource.
+	export.Spec.IsDNSLabelConfigured, err = r.isDNSLabelConfigured(ctx, *id)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *Reconciler) fetchPublicIPResourceID(ctx context.Context, service *corev1.Service) (*string, error) {
+	serviceKObj := klog.KObj(service)
+	lb, err := r.AzureLoadBalancerClient.Get(ctx, r.LoadBalancerResourceGroupName, publicLoadBalancerName, nil)
+	if err != nil {
+		klog.ErrorS(err, "Failed to get Azure load balancer", "service", serviceKObj)
+		return nil, err
+	}
+	frontedIPName := "a" + string(service.UID)
```

Comment:
the dash "-" should be removed: https://github.com/kubernetes-sigs/cloud-provider-azure/blob/6c002772ad81dc56e74a3a726a3b831387989560/pkg/provider/azure_loadbalancer.go#L402, https://github.com/kubernetes/kubernetes/blob/15d08bf7c8813b0533dc147a03d9f42aae735ecd/staging/src/k8s.io/cloud-provider/cloud.go#L89